src/main/java/com/contentful/java/cda/TLSSocketFactory.java

```diff
@@ -0,0 +1,69 @@
+package com.contentful.java.cda;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+/**
+ * Internal class for enabling TLS 1.2 support on https connections, starting from Android 4.4
+ *
+ * {@see https://developer.android.com/reference/javax/net/ssl/SSLSocket.html}
+ */
+final class TLSSocketFactory extends SSLSocketFactory {
+
+  private SSLSocketFactory delegate;
+
+  TLSSocketFactory() throws KeyManagementException, NoSuchAlgorithmException {
+    SSLContext context = SSLContext.getInstance("TLS");
+    context.init(null, null, null);
+    delegate = context.getSocketFactory();
+  }
+
+  @Override
+  public String[] getDefaultCipherSuites() {
+    return delegate.getDefaultCipherSuites();
+  }
+
+  @Override
+  public String[] getSupportedCipherSuites() {
+    return delegate.getSupportedCipherSuites();
+  }
+
+  @Override
+  public Socket createSocket(Socket s, String host, int port, boolean autoClose) throws IOException {
+    return enableTlsOnSocket(delegate.createSocket(s, host, port, autoClose));
+  }
+
+  @Override
+  public Socket createSocket(String host, int port) throws IOException {
+    return enableTlsOnSocket(delegate.createSocket(host, port));
+  }
+
+  @Override
+  public Socket createSocket(String host, int port, InetAddress localHost, int localPort) throws IOException {
+    return enableTlsOnSocket(delegate.createSocket(host, port, localHost, localPort));
+  }
+
+  @Override
+  public Socket createSocket(InetAddress host, int port) throws IOException {
+    return enableTlsOnSocket(delegate.createSocket(host, port));
+  }
+
+  @Override
+  public Socket createSocket(InetAddress address, int port, InetAddress localAddress, int localPort) throws IOException {
+    return enableTlsOnSocket(delegate.createSocket(address, port, localAddress, localPort));
+  }
+
+  private Socket enableTlsOnSocket(Socket socket) {
+    if (socket != null && (socket instanceof SSLSocket)) {
+      ((SSLSocket) socket).setEnabledProtocols(new String[]{"MonkeyBalls"});
```

Comment:
The protocol selected here should be one supported by `socket.getSupportedProtocols` from there, you should choose one that's compatible with TLS 1.2. This way you ensure that every Java version has compatibility.